addon/-legacy-private/system/store.js

```diff
@@ -2840,6 +2897,27 @@ Store = Service.extend({
 
     if (DEBUG) {
       Ember.Test.unregisterWaiter(this.__asyncWaiter);
+      let shouldTrack = this._shouldTrackAsyncRequests;
+      let tracked = this._trackedAsyncRequests;
+      let isSettled = tracked.length === 0;
+
+      if (!isSettled) {
+        if (shouldTrack) {
+          throw new Error(
+            'Async Request leaks detected. Add a breakpoint here and set `store._generateStackTracesForTrackedRequests = true;`to inspect traces for leak origins:\n\t - ' +
+              tracked.map(o => o.label).join('\n\t - ')
+          );
+        } else {
+          warn(
```

Comment:
by default, we may want to warn here. That way apps who violate this, have some grace period to cut over.

Erroring hard, make simply make it hard to upgrade with little upside. That being said, in the future we should strongly consider this being on by default.